pkg/cri/resource-manager/resource-manager.go

```diff
@@ -53,188 +53,290 @@ type resmgr struct {
 
 // NewResourceManager creates a new ResourceManager instance.
 func NewResourceManager() (ResourceManager, error) {
-	var err error
+	m := &resmgr{Logger: logger.NewLogger("resource-manager")}
 
-	if opt.ForceConfig != "" && opt.FallbackConfig != "" {
-		return nil, resmgrError("both fallback (%s) and forced (%s) configuration given",
-			opt.FallbackConfig, opt.ForceConfig)
+	if err := m.setupCache(); err != nil {
+		return nil, err
 	}
 
-	m := &resmgr{
-		Logger: logger.NewLogger("resource-manager"),
+	if err := m.checkConfig(); err != nil {
```

Comment:
Just a thought, should this be named `checkOpts` or smth? The startup configuration management a bit convoluted mechanism and naming could help in understanding the code

Reply:
Hmm, it was called checkOpts in the beginning. The reason for changing it to checkConfig was mostly cosmetic, to have it group better with the other config-related functions. I'll change it back since it looks to be more intuitive.

Reply:
Changed to checkOpts.